lib/vector-core/src/schema/requirement.rs

```diff
@@ -1,9 +1,70 @@
+use std::collections::BTreeMap;
+
+use lookup::LookupBuf;
+use value::{
+    kind::{Collection, Field, Unknown},
+    Kind,
+};
+
+/// The input schema for a given component.
+///
+/// This schema defines the (semantic) fields a component expects to receive from its input
+/// components.
 #[derive(Debug, Clone, PartialEq)]
-pub struct Requirement;
+pub struct Requirement {
+    /// The collection of fields and their types required to be present in the event.
+    ///
+    /// While this can be used to define *exact* requirements on schema fields, it is primarily
+    /// intended for sinks that want to add a type requirement to _all_ fields in the event (e.g.
+    /// JSON encoding).
```

Comment:
I'm not sure how you envision the alternative would look like, could you elaborate on that? Are you considering to remove the ability to restrict exact fields?

Reply:
I've removed this code in 44bc3c1, as it's no longer used since 2af897b landed as well.

We'll probably need to add it back at some point in the future, at which point we can re-evaluate the approach, if needed.